lib/Throttle.php

```diff
@@ -109,17 +110,56 @@ public function applyBruteForcePolicy($uid, $ip) {
 	 * @param string $ip
 	 * @return void
 	 */
-	public function clearSuspiciousAttemptsForUidIpCombination($uid, $ip) {
-		$this->attemptMapper->deleteSuspiciousAttemptsForUidIpCombination($uid, $ip);
+	public function clearFailedLoginAttemptsForUidIpCombination($uid, $ip) {
+		$this->loginAttemptMapper->deleteFailedLoginAttemptsForUidIpCombination($uid, $ip);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @return void
+	 */
+	public function addFailedLinkAccess($token, $ip) {
+		$access = new FailedLinkAccess();
+		$access->setLinkToken($token);
+		$access->setIp($ip);
+		$access->setAttemptedAt($this->timeFactory->getTime());
+		$this->linkAccessMapper->insert($access);
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @throws LinkAuthException
+	 */
+	public function applyBruteForcePolicyForLinkShare($token, $ip) {
+		$banPeriod = $this->config->getBruteForceProtectionBanPeriod();
+		$banUntil = $this->linkAccessMapper->getLastFailedAccessTimeForTokenIpCombination($token, $ip) + $banPeriod;
+		if ($this->linkAccessMapper->getFailedAccessCountForTokenIpCombination($token, $ip) >=
+			$this->config->getBruteForceProtectionFailTolerance() &&
+			$banUntil > $this->timeFactory->getTime()) {
+			throw new LinkAuthException($this->l->t("Too many failed attempts. Try again in %s.",
+				[$this->parseBanPeriodForHumans($banPeriod)])
+			);
+		}
+	}
+
+	/**
+	 * @param string $token
+	 * @param string $ip
+	 * @return void
+	 */
+	public function clearFailedLinkAccesses($token, $ip) {
```

Comment:
Same as above

Reply:
the method removed from class as suggested.